pkg/plugins/golang/multi-module/v1alpha1/api.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```

Comment:
Could you please check the layout of the golang v3 plugin so we follow it here the same approach?

In the root, we would have only the subcommand call (V)
then, we need a dir called scaffolds where we have implementation for them
also, the boilerplates should be protected via an internal dir `[internal](https://github.com/kubernetes-sigs/kubebuilder/tree/1502ede9731d67c591c8a9d3620d6a525ea49953/pkg/plugins/golang/v3/scaffolds/internal)/templates/` so that nobody will import them.

Reply:
I deliberatly removed the scaffold since it cannot be used together with `go edit` commands safely. Right now I do not use any scaffold

Reply:
I think the way that the go.mod is created in the scaffold is not correct since it used a go template. Instead it should probably use either `go mod init` or the package "golang.org/x/mod/modfile".

Reply:
Hi @jakobmoellersap ,

> I deliberatly removed the scaffold since it cannot be used together with go edit commands safely. Right now I do not use any scaffold

What do you mean with?
How create a directory scaffold and have inside of it the "controller (from MVC)" implementation on it can be an issue? 

> I think the way that the go.mod is created in the scaffold is not correct since it used a go template. Instead it should probably use either go mod init or the package "golang.org/x/mod/modfile".

We use the template because we want scaffold / template what should be in the go mod.
You can create a go.mod manually for any Golang application. You do not need to use go mod init at all. That is just a CLI helper.

Reply:
I agree that we could use a scaffolding here and in fact i did this in a previous commit... However I took inspiration from the comment from @everettraven to use the `go edit` commands to also extend them to `go inti` etc. Nevertheless, to stay consistent, I will adjust.

Reply:
@jakobmoellersap you can use go edit that is fine. 
The suggestion is only about how/where you implement the structure of the plugin to ensure that consistency as you say.